common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/ClasspathUtil.xtend

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.dsl.common.util
+
+import java.io.File
+import java.io.FileInputStream
+import java.io.InputStream
+import java.util.ArrayList
+import java.util.List
+import javax.inject.Inject
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.xpath.XPathConstants
+import javax.xml.xpath.XPathFactory
+import org.eclipse.core.runtime.IPath
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.core.runtime.Path
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.jdt.core.IClasspathEntry
+import org.eclipse.jdt.core.JavaCore
+import org.eclipse.xtext.EcoreUtil2
+import org.slf4j.LoggerFactory
+import org.w3c.dom.Document
+import org.w3c.dom.Node
+
+class ClasspathUtil {
```

Comment:
This class has only public methods. Can we decrease the visibility of them?
Also I would prefer separating the logic for Maven and Gradle and not having everything in one class. We could easily extract this and then dispatch to the logic in the method `getBuildToolClasspathEntry`.